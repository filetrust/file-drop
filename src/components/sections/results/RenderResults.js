import React from "react";

import RenderAnalysis from "./RenderAnalysis";
import DownloadFile from "./DownloadFile";
import { IssueMessage, SectionTitle } from "../../widgets";
import DownloadAnalysisReport from "./DownloadAnalysisReport";
import FileAttributes from "./FileAttributes";

function RenderResults({ state }) {
    const { file, analysisReport, analysisReportString, validation } = state;

    if ( validation ) {
        return (
            <div className="validationErrors">
                <p>{validation}</p>
            </div>
        );
    }

    if ( file && analysisReport ) {
        const sanitisations = analysisReport.getElementsByTagName("gw:SanitisationItem");
        const remediations = analysisReport.getElementsByTagName("gw:RemedyItem");
        const issues = analysisReport.getElementsByTagName("gw:IssueItem");
        const [ { value: fileType } = { value: "unknown" } ] = analysisReport.getElementsByTagName("gw:FileType");
        const { name: fileName } = file;
        const hasIssues = issues.length;
        if ( sanitisations.length || remediations.length || hasIssues ) {
            return (
                <div className="analysis">
                    <IssueMessage hasIssues={hasIssues}/>
                    <SectionTitle context='regenerated'>Your Safe, Regenerated File Is Ready</SectionTitle>
                    <div className="analysis-container buttons-container touch-full">
                        <DownloadFile file={file} hasIssues={hasIssues}/>
                        <DownloadAnalysisReport report={analysisReportString} filename={fileName}/>
                    </div>

                    <FileAttributes file={file} fileType={fileType}/>

                    <RenderAnalysis
                        remediations={remediations}
                        sanitisations={sanitisations}
                        issues={issues}
                    />
                </div>
            );
        } else {
            return (
            <div className="is-clean analysis">
                <SectionTitle context='clean'>File is clean!</SectionTitle>
                <div className="download-container buttons-container touch-full">
                    <DownloadAnalysisReport report={analysisReportString} filename={fileName}/>
                </div>
                <FileAttributes file={file} fileType={fileType}/>
            </div>
            );
        }
    }

    return null;
}

export default RenderResults;
