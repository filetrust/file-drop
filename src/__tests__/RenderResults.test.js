import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import RenderResults from "../components/sections/results/RenderResults";

Enzyme.configure({adapter: new Adapter()});

const validation = "File Size is too big";

describe("RenderResults", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<RenderResults />);
    expect(wrapper.exists()).toBe(true);
  });

  it("matches snapshot", () => {
    const wrapper = shallow(<RenderResults />);
    expect(wrapper).toMatchSnapshot();
  });

  it("displays validation errors if validation exists", () => {
    const wrapper = shallow(<RenderResults state={{ validation: validation }} />);
    expect(
      wrapper.contains(
        <div className="validationErrors">
          <p>{validation}</p>
        </div>
      )
    ).toBe(true);
  });

  it("doesn't display validation errors if no validation exists", () => {
    const wrapper = shallow(<RenderResults state={{ validation: null }}  />);
    expect(
      wrapper.contains(
        <div className="validationErrors">
          <p>{validation}</p>
        </div>
      )
    ).toBe(false);
  });
});
