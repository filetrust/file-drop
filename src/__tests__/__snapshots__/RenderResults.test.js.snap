// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderResults matches snapshot 1`] = `ShallowWrapper {}`;
